### DISCUSSION TITLE
Using custom DNS for cluster endpoint

### DISCUSSION THREAD

**Discussion by cmvagner** [awslabs/aws-mysql-jdbc#58] [Q&A] [answered]
Hi,
We use custom DNS for cluster endpoints (regular and read-only) for being able to switch the behind database without affecting the clients.
So for example "db-sandbox.customDomain.com" points to cluster endpoint "sandbox.cluster-_account-identifier_._region_.rds.amazonaws.com".
The cluster consists of a Writer and one Reader. How should I configure the driver to handle failover using clusterInstanceHostPattern?
Should I use "?.cluster-_account-identifier_._region_.rds.amazonaws.com" as clusterInstanceHostPattern or am I misunderstanding something?

**Comment by sergiyvamz** [comment, marked as answer]
Hi,
You can use "?.account-identifier.region.rds.amazonaws.com" (not "?.cluster-account-identifier.region.rds.amazonaws.com" as you've mentioned) as clusterInstanceHostPattern. Another option is to use "?.customDomain.com" if your DNS can resolve it to db instance endpoints or corresponding IP addresses. 

Here's an example. Let's assume your db cluster has two instances ("instance-a" and "instance-b"). 
1) If you use  "clusterInstanceHostPattern=?.account-identifier.region.rds.amazonaws.com", the question mark will be replaced by actual instance name:
"instance-a" -> "instance-a.account-identifier.region.rds.amazonaws.com"
"instance-b" -> "instance-b.account-identifier.region.rds.amazonaws.com"
2) If you use  "clusterInstanceHostPattern=?.customDomain.com", the question mark will be replaced by actual instance name:
"instance-a" -> "instance-a.customDomain.com"
"instance-b" -> "instance-b.customDomain.com"
In this case, your DNS should be properly configured to resolve such endpoints to cluster instances.

I hope that clarifies your case.

**Reply by cmvagner** [reply]
Great,  thanks for answering.
I have created CNAME records for both my instances in the cluster and is then using clusterInstanceHostPattern like "clusterInstanceHostPattern=?.customDomain.com".
I do though have an issue getting failover to work using custom domain. In the failover process it will end up connecting to the reader (which was the writer before) and will then fail because of --read-only flag. I have no failover issues just using the cluster endpoint. 
Its not directly related to this question so I should properly create this as a separate issue.

**Reply by sergiyvamz** [reply]
I believe that issue has already been fixed, but isn't yet included in the latest release and in the published jar. The next release, (coming soon), will include this fix and the driver will properly handle the case you described. In the meantime, I'd suggest following our instructions to build the driver from source.
